kod/object/item.kod

```diff
@@ -633,6 +690,46 @@ messages:
       return FALSE;
    }
 
+   GetMagicRarityGradeCount()
+   "Checks plItem_attributes for HasRarity() = TRUE."
+   "For each attribute with HasRarity = TRUE,  add 1 to count"
+   "Return either viRarity or count, whichever is larger"
+   "This ensures the higher rarity is chosen if in the case "
+   "of competing rarities for an item."
+   {
+
+      local i, iNum, oItemAtt, count;
+      count = 0;
+
+      if Send(self,@HasAnyAttribute)
+      {
+         for i in plItem_attributes
+         {
+            iNum = Send(self,@GetNumFromCompound,#compound=First(i));
+            oItemAtt = Send(SYS,@FindItemAttByNum,#num=iNum);
+
+            if Send(oItemAtt,@HasRarity)
+            {
+               count = count + 1;
+            }
+         }
+      }
+
+      % Use assigned class rarity grade if it is higher than the count
+      if viRarity > count
```

Comment:
I'd use Bound() here and then have only one return statement.  Less fragile if code is added in the future.

Reply:
Done.  

However, I had to use the format of `bound(count, viRarity, $)`

From what I read in the manual, it looks like it should not be in this format.  It should be:
`bound(n ,a, b) = bound(count, $, viRarity) = max( min(count, $), viRarity)`

I do want `max(n, b)` which is `max(count, viRarity)` so I am slightly bothered why that doesn't work.

But it seems to work from my change above in testing.

>Bound: Takes an integer (call it n) and two values which can be integers or nil. Call the
first of these a and the second b. If a and b are both nil, then this function returns n.
If a is nil and b is an integer, then it returns max(n,b). If a is an integer and b is nil,
then it returns min(n,a). If both a and b are integers, returns max(min(n,a),b).

Reply:
You are correct and the manual is wrong.  The manual source has been fixed, but I haven't gotten around to regenerating the PDF of it (which is a hassle since it's LaTeX).